src/io_uring.rs

```diff
@@ -947,6 +1071,18 @@ impl Default for addr_or_splice_off_in_union {
     }
 }
 
+impl Default for addr3_or_cmd_union {
+    #[inline]
+    fn default() -> Self {
+        let mut s = ::core::mem::MaybeUninit::<Self>::uninit();
+        // Safety: All of Linux's io_uring structs may be zero-initialized.
+        unsafe {
+            ::core::ptr::write_bytes(s.as_mut_ptr(), 0, 1);
```

Comment:
(Hmm, does LLVM really not optimize the case of the "default `default()`" function to this zero memset?)

Reply:
The `Default` derive doesn't work for union types, so we need to provide our own implementation.